src/Connections/GenericHook.ts

```diff
@@ -258,18 +264,41 @@ export class GenericHookConnection extends BaseConnection implements IConnection
         }
     }
 
-    public async onStateUpdate(stateEv: MatrixEvent<unknown>) {
-        const validatedConfig = GenericHookConnection.validateState(stateEv.content as Record<string, unknown>, this.config.allowJsTransformationFunctions || false);
-        if (validatedConfig.transformationFunction) {
-            try {
-                this.transformationFunction = new Script(validatedConfig.transformationFunction);
-            } catch (ex) {
-                await this.messageClient.sendMatrixText(this.roomId, 'Could not compile transformation function:' + ex, "m.text", this.intent.userId);
-            }
+    public onStateUpdate(stateEv: MatrixEvent<unknown>) {
+        return this.updateConfig(stateEv.content as Record<string, unknown>);
+    }
+
+    private async updateConfig(config: Record<string, unknown>) {
+        this.state = {
+            ...GenericHookConnection.validateState(config),
+            hookId: this.state.hookId,
+        };
+        return this.validateTransformationFunction();
+    }
+
+    private validateTransformationFunction(): Promise<void> {
+        let result = Promise.resolve();
+        if (!GenericHookConnection.quickModule || !this.state.transformationFunction) {
+            return result;
+        }
+        this.hasValidTransformation = false;
+        const ctx = GenericHookConnection.quickModule.newContext();
+        const codeEvalResult = ctx.evalCode(`function f(data) {${this.state.transformationFunction}}`);
+        if (codeEvalResult.error) {
+            const message = "Could not compile transformation function:\n```" + JSON.stringify(ctx.dump(codeEvalResult.error)) + "```";
+            codeEvalResult.error.dispose();
+            result = this.intent.sendEvent(this.roomId, {
+                msgtype: "m.text",
+                body: message,
+                formatted_body: md.renderInline(message).replaceAll("\n", "<br>"),
```

Comment:
I'd hope that renderInline would aleady be using HTML elements...

Reply:
I have a commit that fixes this.  Luckily the answer was just a few lines below :slightly_smiling_face:

Reply:
Fixed, it now looks like this:
![image](https://github.com/matrix-org/matrix-hookshot/assets/3271094/809596c0-2363-404a-965d-d6de7c5fcfb0)